cluster/manifests/01-admission-control/config.yaml

```diff
@@ -7,7 +7,8 @@ data:
   daemonset.reserved.cpu: "{{ .Cluster.ConfigItems.teapot_admission_controller_daemonset_reserved_cpu }}"
   daemonset.reserved.memory: "{{ .Cluster.ConfigItems.teapot_admission_controller_daemonset_reserved_memory }}"
 
-  dns.default.hostname-max-length: "{{ .Cluster.ConfigItems.hostname_max_length }}"
+  dns.default.subdomain-max-length: "{{ .Cluster.ConfigItems.subdomain_max_length }}"
+  dns.default.hostname-max-length: "{{ .Cluster.ConfigItems.hostname_max_length }}" # to be removed
```

Comment:
As @mikkeloscar already suggested I also believe this needs to be rolled out in a separate PR before this one.

This first thing that happens is updating the userdata.yaml below which will roll the master nodes. The masters will run `admission-controller:master-157` which is looking for the key `subdomain-max-length` in the admission config map. However, that config map hasn't been updated yet because that happens after. It's only a windows of like 20 minutes if you don't roll any workers, too, but that's too long.

Separating the userdata change into another PR should fix it. And I think you can also change this part here to
```
dns.default.subdomain-max-length: "{{ .Cluster.ConfigItems.subdomain_max_length }}"
dns.default.hostname-max-length: "{{ .Cluster.ConfigItems.subdomain_max_length }}" # to be removed
```
to safe one config item (but i'm not sure, your solution is definitely safer in that regard).

Reply:
Sounds good. Keep in mind that it needs to be split this way for every rollout to the other channels as well. This requires a little bit of coordination and we can easily forget it.